pcsx2/GS/Renderers/HW/GSTextureCache.h

```diff
@@ -181,6 +181,34 @@ class GSTextureCache
 		bool operator()(const PaletteKey& lhs, const PaletteKey& rhs) const;
 	};
 
+	class Target : public Surface
+	{
+	public:
+		const int m_type = 0;
+		bool m_used = false;
+		GSDirtyRectList m_dirty;
+		GSVector4i m_valid{};
+		GSVector4i m_drawn_since_read{};
+		const bool m_depth_supported = false;
+		bool m_dirty_alpha = true;
+		bool m_is_frame = false;
+		int readbacks_since_draw = 0;
+
+	public:
+		Target(const GIFRegTEX0& TEX0, const bool depth_supported, const int type);
+		~Target();
+
+		void UpdateValidity(const GSVector4i& rect);
+
+		void Update(bool reset_age, GIFRegTEX0 TEX0);
+
+		/// Updates the target, if the dirty area intersects with the specified rectangle.
+		void UpdateIfDirtyIntersects(const GSVector4i& rc);
+
+		bool ResizeTexture(int new_width, int new_height, bool recycle_old = true);
```

Comment:
The ResizeTexture functions have the same name, but they have different arguments. This could lead to confusion if the wrong function is called by mistake. It would be better to give them distinct names to make it clear which one to use for a given situation.